From 9ab441fe2e7b692b0d9adcfe53c10673535caa21 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Tue, 4 Jun 2024 11:04:44 +0200
Subject: [PATCH] ARM: dts: stm32: Replace OOB interrupt by IB interrupt for
 WIFI on stm32mp135f-dk

Enabling SDIO IRQ signalling on this interface instead of out-of-band
interrupt from brcmfmac

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Signed-off-by: Erwan SZYMANSKI <erwan.szymanski@st.com>
Change-Id: I0ca51bbdbe08e5c96d786f4f480d018a0786ee53
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/385028
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp135f-dk.dts | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp135f-dk.dts b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
index e0cd6e1363f6..5ce9f62c172d 100644
--- a/arch/arm/boot/dts/st/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
@@ -489,6 +489,7 @@ &sdmmc2 {
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a &sdmmc2_clk_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	non-removable;
+	cap-sdio-irq;
 	st,neg-edge;
 	bus-width = <4>;
 	vmmc-supply = <&v3v3_ao>;
@@ -500,9 +501,6 @@ &sdmmc2 {
 	brcmf: bcrmf@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
-		interrupt-parent = <&gpiof>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>; /* WL_HOST_WAKE */
-		interrupt-names = "host-wake";
 	};
 };
 
-- 
2.39.5

